tests/Incoming/Test/Structure/StructureFactoryTest.php

```diff
@@ -122,4 +122,17 @@ public function testBuildWithInvalidStructuralType()
 
         $structure = (new StructureFactory)->build($data);
     }
+    
+    /**
+     * @see https://github.com/Rican7/incoming/pull/3
+     */
+    public function testBuildWithTraversableAndMixedKeys()
+    {
+        $data = ['name' => 'markus', 0 => 'a crazy mixed in key'];
+
+        $structure = (new StructureFactory)->build(new ArrayIterator($data));
+
+        $this->assertInstanceOf('Incoming\Structure\Map', $structure);
```

Comment:
@Rican7 btw: you should use `assertInstanceOf` instead of `assertTrue($x instanceof MyClass)` also in other tests.